File: src/main/java/com/exampleaaaa/demo/LearningCollection.java
package com.exampleaaaa.demo;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;

@Entity
@Table(name = "LearningCollection")
@Data
public class LearningCollection implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @Column
    String learning;

    @Column
    String category;

    @Column
    LocalDate dateAdded;

}

File: src/main/kotlin/com/menketechnologies/learningcollection/LearningCollection.kt
package com.menketechnologies.learningcollection

import java.io.Serializable
import java.util.*
import jakarta.persistence.*

@Entity
@Table(name = "LearningCollection")
class LearningCollection : Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column
    var id: Long = 0

    @Column
    lateinit var learning: String

    @Column
    lateinit var category: String

    @Column
    lateinit var dateAdded: Date


    constructor(learning: String, category: String, dateAdded: Date) {
        this.learning = learning
        this.category = category
        this.dateAdded = dateAdded
    }

    constructor()


    companion object {
        private const val serialVersionUID = 1L
    }
}
